<a href="https://colab.research.google.com/github/SergioShujieHuang/gamebuzz_jp/blob/main/experiment1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# google mount

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# import package

In [3]:
import pandas as pd

In [4]:
!pwd

/content


In [5]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

In [6]:
df = pd.read_excel('./drive/MyDrive/file_backup/Master/研究/実験１/Book1.xlsx', header=1)
df = df.set_index('number')

In [7]:
display(df)

,原文読む時間,要約読む時間,テスト時間,誤りを気づいた,気づいた数,気づいた誤り１,気づいた誤り２,事前知識,Total score,太陽光が球状の雨粒に入り、虹ができるまでの光の経路として、正しい順序はどれか,太陽光が球状の雨粒に入り、虹ができるまでの光の経路として、正しい順序はどれか [Score],虹が「色の帯」として見えるのは、光の「ある性質」が異なることで屈折角（曲がり方）が変わるためである。その「ある性質」とは何か,虹が「色の帯」として見えるのは、光の「ある性質」が異なることで屈折角（曲がり方）が変わるためである。その「ある性質」とは何か [Score],光の色（波長）と屈折角の関係について、正しいものはどれか,光の色（波長）と屈折角の関係について、正しいものはどれか [Score],なぜ虹は直線や不規則な形ではなく「弧（アーチ状）」に見えるのか。その理由に関わる説明として正しいものはどれか,なぜ虹は直線や不規則な形ではなく「弧（アーチ状）」に見えるのか。その理由に関わる説明として正しいものはどれか [Score],地上の観察者から見て、主虹（普通の虹）が最も強く光を散乱する角度は、太陽の方向からおよそ何度か,地上の観察者から見て、主虹（普通の虹）が最も強く光を散乱する角度は、太陽の方向からおよそ何度か [Score],虹が完全な円ではなく「弧」に見える理由として、要因はどれか,虹が完全な円ではなく「弧」に見える理由として、要因はどれか [Score],「二重虹（外側の虹）」が形成される際、光は雨粒の内部で何回反射するか,「二重虹（外側の虹）」が形成される際、光は雨粒の内部で何回反射するか [Score],二重虹（外側の虹）の色の並び順について、正しいものはどれか,二重虹（外側の虹）の色の並び順について、正しいものはどれか [Score],二重虹（外側の虹）の明るさと角度について、通常の虹と比較したときの特徴はどれか,二重虹（外側の虹）の明るさと角度について、通常の虹と比較したときの特徴はどれか [Score],虹の明るさや色の鮮明さが変化する要因として挙げられているものはどれか,虹の明るさや色の鮮明さが変化する要因として挙げられているものはどれか [Score]
number,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
1,2m22s9,0m28s99,2m47s30,F,NaN,NaN,NaN,なし,50.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の強度（明るさ）,0.00 / 10,長波長（赤）の方が、短波長（青・紫）よりも強く曲がる,0.00 / 10,雨粒内での反射によって光が方向を変えるため,0.00 / 10,約 40〜42°,10.00 / 10,太陽光が大気中で不均一に減衰するため,0.00 / 10,2回,10.00 / 10,通常の虹とは色が反転している,10.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,水滴内での光の反射回数,0.00 / 10
2,3m14s23,1m43s06,3m02s95,F,NaN,NaN,NaN,なし,70.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の波長,10.00 / 10,短波長（青・紫）の方が、長波長（赤）よりも強く曲がる,10.00 / 10,一定角度条件を満たす雨粒が空間的に分布するため,10.00 / 10,約 40〜42°,10.00 / 10,観察可能な雨粒の空間範囲が幾何的に制限されるため,10.00 / 10,1回,0.00 / 10,青色のみが見えなくなり、赤色だけが残る,0.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,太陽光の波長成分の数,0.00 / 10
3,3m43s02,2m16s29,2m27s40,T,1.0,T,NaN,なし,60.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の波長,10.00 / 10,長波長（赤）の方が、短波長（青・紫）よりも強く曲がる,0.00 / 10,雨粒内での反射によって光が方向を変えるため,0.00 / 10,約 40〜42°,10.00 / 10,観察可能な雨粒の空間範囲が幾何的に制限されるため,10.00 / 10,2回,10.00 / 10,通常の虹と同じ並び順である,0.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,水滴内での光の反射回数,0.00 / 10
4,3m22s59,2m03s97,4m21s00,F,NaN,NaN,NaN,なし,40.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の入射角,0.00 / 10,長波長（赤）の方が、短波長（青・紫）よりも強く曲がる,0.00 / 10,一定角度条件を満たす雨粒が空間的に分布するため,10.00 / 10,約 40〜42°,10.00 / 10,雨粒ごとに光の屈折条件がわずかに異なるため,0.00 / 10,2回,10.00 / 10,通常の虹と同じ並び順である,0.00 / 10,明るさは弱く、角度はより小さい,0.00 / 10,水滴内での光の反射回数,0.00 / 10
5,2m45s69,1m17s76,2m08s52,F,NaN,NaN,NaN,なし,80.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の波長,10.00 / 10,長波長（赤）の方が、短波長（青・紫）よりも強く曲がる,0.00 / 10,一定角度条件を満たす雨粒が空間的に分布するため,10.00 / 10,約 40〜42°,10.00 / 10,観察可能な雨粒の空間範囲が幾何的に制限されるため,10.00 / 10,2回,10.00 / 10,通常の虹とは色が反転している,10.00 / 10,明るさは強く、角度はより大きい,0.00 / 10,大気中の光散乱と雨粒の大きさ,10.00 / 10
6,2m16s35,1m26.77,2m57s05,T,2.0,T,T,あり,80.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の波長,10.00 / 10,短波長（青・紫）の方が、長波長（赤）よりも強く曲がる,10.00 / 10,一定角度条件を満たす雨粒が空間的に分布するため,10.00 / 10,約 40〜42°,10.00 / 10,観察者の位置によって可視角度が変化するため,0.00 / 10,2回,10.00 / 10,通常の虹とは色が反転している,10.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,水滴内での光の反射回数,0.00 / 10
7,2m02s09,0m28s19,2m27s91,F,NaN,NaN,NaN,なし,60.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の波長,10.00 / 10,長波長（赤）の方が、短波長（青・紫）よりも強く曲がる,0.00 / 10,雨粒内での反射によって光が方向を変えるため,0.00 / 10,約 40〜42°,10.00 / 10,観察者の位置によって可視角度が変化するため,0.00 / 10,2回,10.00 / 10,通常の虹と同じ並び順である,0.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,大気中の光散乱と雨粒の大きさ,10.00 / 10
8,5m44s03,3m12s19,2m17s90,F,NaN,NaN,NaN,なし,80.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の波長,10.00 / 10,長波長（赤）の方が、短波長（青・紫）よりも強く曲がる,0.00 / 10,一定角度条件を満たす雨粒が空間的に分布するため,10.00 / 10,約 40〜42°,10.00 / 10,観察者の位置によって可視角度が変化するため,0.00 / 10,2回,10.00 / 10,通常の虹とは色が反転している,10.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,大気中の光散乱と雨粒の大きさ,10.00 / 10
9,2m54s52,0m38s53,1m33s22,F,NaN,NaN,NaN,なし,70.00 / 100,雨粒に入る（屈折）→ 内壁で透過→ 雨粒から出る（屈折）,0.00 / 10,光の波長,10.00 / 10,短波長（青・紫）の方が、長波長（赤）よりも強く曲がる,10.00 / 10,一定角度条件を満たす雨粒が空間的に分布するため,10.00 / 10,約 40〜42°,10.00 / 10,雨粒ごとに光の屈折条件がわずかに異なるため,0.00 / 10,2回,10.00 / 10,通常の虹と同じ並び順である,0.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,大気中の光散乱と雨粒の大きさ,10.00 / 10


# 事前知识ありを除外

In [8]:
df = df[df['事前知識'] == 'なし']
display(df)

新的DataFrame (df_なし) 的前5行数据如下：


,原文読む時間,要約読む時間,テスト時間,誤りを気づいた,気づいた数,気づいた誤り１,気づいた誤り２,事前知識,Total score,太陽光が球状の雨粒に入り、虹ができるまでの光の経路として、正しい順序はどれか,太陽光が球状の雨粒に入り、虹ができるまでの光の経路として、正しい順序はどれか [Score],虹が「色の帯」として見えるのは、光の「ある性質」が異なることで屈折角（曲がり方）が変わるためである。その「ある性質」とは何か,虹が「色の帯」として見えるのは、光の「ある性質」が異なることで屈折角（曲がり方）が変わるためである。その「ある性質」とは何か [Score],光の色（波長）と屈折角の関係について、正しいものはどれか,光の色（波長）と屈折角の関係について、正しいものはどれか [Score],なぜ虹は直線や不規則な形ではなく「弧（アーチ状）」に見えるのか。その理由に関わる説明として正しいものはどれか,なぜ虹は直線や不規則な形ではなく「弧（アーチ状）」に見えるのか。その理由に関わる説明として正しいものはどれか [Score],地上の観察者から見て、主虹（普通の虹）が最も強く光を散乱する角度は、太陽の方向からおよそ何度か,地上の観察者から見て、主虹（普通の虹）が最も強く光を散乱する角度は、太陽の方向からおよそ何度か [Score],虹が完全な円ではなく「弧」に見える理由として、要因はどれか,虹が完全な円ではなく「弧」に見える理由として、要因はどれか [Score],「二重虹（外側の虹）」が形成される際、光は雨粒の内部で何回反射するか,「二重虹（外側の虹）」が形成される際、光は雨粒の内部で何回反射するか [Score],二重虹（外側の虹）の色の並び順について、正しいものはどれか,二重虹（外側の虹）の色の並び順について、正しいものはどれか [Score],二重虹（外側の虹）の明るさと角度について、通常の虹と比較したときの特徴はどれか,二重虹（外側の虹）の明るさと角度について、通常の虹と比較したときの特徴はどれか [Score],虹の明るさや色の鮮明さが変化する要因として挙げられているものはどれか,虹の明るさや色の鮮明さが変化する要因として挙げられているものはどれか [Score]
number,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
1,2m22s9,0m28s99,2m47s30,F,NaN,NaN,NaN,なし,50.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の強度（明るさ）,0.00 / 10,長波長（赤）の方が、短波長（青・紫）よりも強く曲がる,0.00 / 10,雨粒内での反射によって光が方向を変えるため,0.00 / 10,約 40〜42°,10.00 / 10,太陽光が大気中で不均一に減衰するため,0.00 / 10,2回,10.00 / 10,通常の虹とは色が反転している,10.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,水滴内での光の反射回数,0.00 / 10
2,3m14s23,1m43s06,3m02s95,F,NaN,NaN,NaN,なし,70.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の波長,10.00 / 10,短波長（青・紫）の方が、長波長（赤）よりも強く曲がる,10.00 / 10,一定角度条件を満たす雨粒が空間的に分布するため,10.00 / 10,約 40〜42°,10.00 / 10,観察可能な雨粒の空間範囲が幾何的に制限されるため,10.00 / 10,1回,0.00 / 10,青色のみが見えなくなり、赤色だけが残る,0.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,太陽光の波長成分の数,0.00 / 10
3,3m43s02,2m16s29,2m27s40,T,1.0,T,NaN,なし,60.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の波長,10.00 / 10,長波長（赤）の方が、短波長（青・紫）よりも強く曲がる,0.00 / 10,雨粒内での反射によって光が方向を変えるため,0.00 / 10,約 40〜42°,10.00 / 10,観察可能な雨粒の空間範囲が幾何的に制限されるため,10.00 / 10,2回,10.00 / 10,通常の虹と同じ並び順である,0.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,水滴内での光の反射回数,0.00 / 10
4,3m22s59,2m03s97,4m21s00,F,NaN,NaN,NaN,なし,40.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の入射角,0.00 / 10,長波長（赤）の方が、短波長（青・紫）よりも強く曲がる,0.00 / 10,一定角度条件を満たす雨粒が空間的に分布するため,10.00 / 10,約 40〜42°,10.00 / 10,雨粒ごとに光の屈折条件がわずかに異なるため,0.00 / 10,2回,10.00 / 10,通常の虹と同じ並び順である,0.00 / 10,明るさは弱く、角度はより小さい,0.00 / 10,水滴内での光の反射回数,0.00 / 10
5,2m45s69,1m17s76,2m08s52,F,NaN,NaN,NaN,なし,80.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の波長,10.00 / 10,長波長（赤）の方が、短波長（青・紫）よりも強く曲がる,0.00 / 10,一定角度条件を満たす雨粒が空間的に分布するため,10.00 / 10,約 40〜42°,10.00 / 10,観察可能な雨粒の空間範囲が幾何的に制限されるため,10.00 / 10,2回,10.00 / 10,通常の虹とは色が反転している,10.00 / 10,明るさは強く、角度はより大きい,0.00 / 10,大気中の光散乱と雨粒の大きさ,10.00 / 10
7,2m02s09,0m28s19,2m27s91,F,NaN,NaN,NaN,なし,60.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の波長,10.00 / 10,長波長（赤）の方が、短波長（青・紫）よりも強く曲がる,0.00 / 10,雨粒内での反射によって光が方向を変えるため,0.00 / 10,約 40〜42°,10.00 / 10,観察者の位置によって可視角度が変化するため,0.00 / 10,2回,10.00 / 10,通常の虹と同じ並び順である,0.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,大気中の光散乱と雨粒の大きさ,10.00 / 10
8,5m44s03,3m12s19,2m17s90,F,NaN,NaN,NaN,なし,80.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の波長,10.00 / 10,長波長（赤）の方が、短波長（青・紫）よりも強く曲がる,0.00 / 10,一定角度条件を満たす雨粒が空間的に分布するため,10.00 / 10,約 40〜42°,10.00 / 10,観察者の位置によって可視角度が変化するため,0.00 / 10,2回,10.00 / 10,通常の虹とは色が反転している,10.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,大気中の光散乱と雨粒の大きさ,10.00 / 10
9,2m54s52,0m38s53,1m33s22,F,NaN,NaN,NaN,なし,70.00 / 100,雨粒に入る（屈折）→ 内壁で透過→ 雨粒から出る（屈折）,0.00 / 10,光の波長,10.00 / 10,短波長（青・紫）の方が、長波長（赤）よりも強く曲がる,10.00 / 10,一定角度条件を満たす雨粒が空間的に分布するため,10.00 / 10,約 40〜42°,10.00 / 10,雨粒ごとに光の屈折条件がわずかに異なるため,0.00 / 10,2回,10.00 / 10,通常の虹と同じ並び順である,0.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,大気中の光散乱と雨粒の大きさ,10.00 / 10
10,3m16s19,1m10s82,3m37s88,T,2.0,T,T,なし,100.00 / 100,雨粒に入る（屈折）→ 内壁で反射→ 雨粒から出る（屈折）,10.00 / 10,光の波長,10.00 / 10,短波長（青・紫）の方が、長波長（赤）よりも強く曲がる,10.00 / 10,一定角度条件を満たす雨粒が空間的に分布するため,10.00 / 10,約 40〜42°,10.00 / 10,観察可能な雨粒の空間範囲が幾何的に制限されるため,10.00 / 10,2回,10.00 / 10,通常の虹とは色が反転している,10.00 / 10,明るさは弱く、角度はより大きい,10.00 / 10,大気中の光散乱と雨粒の大きさ,10.00 / 10


# Task
Perform a logistic regression analysis using the data in `df` to determine if the proportion of time spent reading the original text affects the likelihood of noticing errors.

First, preprocess the data from "./drive/MyDrive/file_backup/Master/研究/実験１/Book1.xlsx":
1. Convert time columns (e.g., '原文読む時間', '要約読む時間') from 'XXmXXsXX' format to total seconds.
2. Calculate the 'proportion of time spent reading the original text' relative to the total reading time (original + summary).
3. Convert the '誤りを気づいた' column from 'T/F' to binary (1/0).

Next, execute a logistic regression using the `statsmodels` library with '誤りを気づいた' as the dependent variable and the 'time proportion' as the independent variable.

Finally, output the results including the Wald Z-test statistics, confidence intervals, and the effect size (odds ratio), accompanied by an interpretation of the statistical significance in Japanese.

## data_preprocessing

### Subtask:
Convert time columns to seconds, calculate the original reading time proportion, and convert the target variable to binary.


In [9]:
import re

def parse_time_to_seconds(time_str):
    if pd.isna(time_str) or not isinstance(time_str, str):
        return 0.0
    # Matches formats like '2m22s9' or '0m28s99'
    match = re.match(r'(\d+)m(\d+)s(\d+)', time_str)
    if match:
        minutes = int(match.group(1))
        seconds = int(match.group(2))
        centiseconds = int(match.group(3))
        # Convert everything to seconds
        return minutes * 60 + seconds + centiseconds / 100.0
    return 0.0

# 1 & 2. Convert time columns to numerical seconds
df['原文読む時間_sec'] = df['原文読む時間'].apply(parse_time_to_seconds)
df['要約読む時間_sec'] = df['要約読む時間'].apply(parse_time_to_seconds)

# 3. Calculate original reading time proportion
df['original_reading_proportion'] = df['原文読む時間_sec'] / (df['原文読む時間_sec'] + df['要約読む時間_sec'])

# 4. Map '誤りを気づいた' to binary (T=1, F=0)
df['誤りを気づいた_binary'] = df['誤りを気づいた'].map({'T': 1, 'F': 0}).astype(float)

# 5. Inspect the result
print("Preprocessed DataFrame:")
display(df[['原文読む時間', '原文読む時間_sec', '要約読む時間', '要約読む時間_sec', 'original_reading_proportion', '誤りを気づいた_binary']].head())

Preprocessed DataFrame:


,原文読む時間,原文読む時間_sec,要約読む時間,要約読む時間_sec,original_reading_proportion,誤りを気づいた_binary
number,,,,,,
1,2m22s9,142.09,0m28s99,28.99,0.830547,0.0
2,3m14s23,194.23,1m43s06,103.06,0.653335,0.0
3,3m43s02,223.02,2m16s29,136.29,0.620690,1.0
4,3m22s59,202.59,2m03s97,123.97,0.620376,0.0
5,2m45s69,165.69,1m17s76,77.76,0.680591,0.0


## logistic_regression_analysis

### Subtask:
Perform a logistic regression using statsmodels to analyze the impact of original reading time proportion on error detection.


In [10]:
import statsmodels.api as sm
import numpy as np

# 1. Define independent (X) and dependent (y) variables
# Adding a constant for the intercept term
X = sm.add_constant(df['original_reading_proportion'])
y = df['誤りを気づいた_binary']

# 2. Fit the logistic regression model
model = sm.Logit(y, X)
results = model.fit()

# 3. Calculate Odds Ratios and Confidence Intervals
odds_ratios = np.exp(results.params)
conf_int = results.conf_int()
odds_conf_int = np.exp(conf_int)

# 4. Consolidate results for interpretation
odds_df = pd.DataFrame({
    'Odds Ratio': odds_ratios,
    'Lower CI (95%)': odds_conf_int[0],
    'Upper CI (95%)': odds_conf_int[1]
})

# 5. Output the results
print(results.summary())
print("\n--- Odds Ratios and Confidence Intervals ---")
display(odds_df)

# Japanese interpretation based on p-value
p_value = results.pvalues['original_reading_proportion']
if p_value < 0.05:
    print(f"\n統計的有意性: 有意水準5%で、原文読む時間の割合が誤りを気づく確率に有意な影響を与えていると言えます (p = {p_value:.4f})。")
else:
    print(f"\n統計的有意性: 有意水準5%で、原文読む時間の割合が誤りを気づく確率に有意な影響を与えているとは言えません (p = {p_value:.4f})。")

Optimization terminated successfully.
         Current function value: 0.558206
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:         誤りを気づいた_binary   No. Observations:                   12
Model:                          Logit   Df Residuals:                       10
Method:                           MLE   Df Model:                            1
Date:                Wed, 25 Feb 2026   Pseudo R-squ.:                0.007343
Time:                        01:52:52   Log-Likelihood:                -6.6985
converged:                       True   LL-Null:                       -6.7480
Covariance Type:            nonrobust   LLR p-value:                    0.7529
                                  coef    std err          z      P>|z|      [0.025      0.975]
-----------------------------------------------------------------------------------------------
const                          -2.4738      4.550     -0.544      0.587   

,Odds Ratio,Lower CI (95%),Upper CI (95%)
const,0.084262,0.000011,6.286245e+02
original_reading_proportion,7.239680,0.000025,2.123076e+06



統計的有意性: 有意水準5%で、原文読む時間の割合が誤りを気づく確率に有意な影響を与えているとは言えません (p = 0.7579)。


## interpretation_and_results

### Subtask:
Summarize the logistic regression results including Wald Z-statistics, odds ratios, and Japanese interpretation.


## Summary:

### Q&A

**Does the proportion of time spent reading the original text affect the likelihood of noticing errors?**
Based on the logistic regression analysis, the proportion of time spent reading the original text does not have a statistically significant effect on the likelihood of noticing errors ($p = 0.7579$).

### Data Analysis Key Findings

*   **Time Data Preprocessing**: Time strings in the format 'XXmXXsXX' were successfully converted to numerical seconds. For example, '2m22s9' was parsed as **142.09 seconds**.
*   **Model Coefficients**: The logistic regression yielded a coefficient of **1.9796** for the original reading proportion, with a Wald Z-statistic of **0.308**.
*   **Odds Ratio**: The odds ratio was calculated at **7.240**. However, the 95% confidence interval was extremely wide (**0.000025 to 2,123,076**), suggesting high instability in the estimate.
*   **Statistical Significance**: The p-value of **0.7579** is well above the standard 0.05 threshold. Therefore, we cannot conclude that reading time proportion influences error detection.
*   **Interpretation (Japanese)**: 統計的有意性: 有意水準5%で、原文読む時間の割合が誤りを気づく確率に有意な影響を与えているとは言えません (p = 0.7579)。

### Insights or Next Steps

*   **Increase Sample Size**: The extremely wide confidence intervals and high p-value are likely driven by the very small sample size ($N=12$). Future experiments should involve more participants to improve the power of the statistical test.
*   **Control for Other Variables**: Since time proportion was not significant, other factors such as reading comprehension levels or the specific nature of the errors should be explored as potential predictors for noticing errors.


# Task
Perform a mixed-design ANCOVA (Analysis of Covariance) to evaluate the impact of noticing errors on test performance using the data from "./drive/MyDrive/file_backup/Master/研究/実験１/Book1.xlsx".

Specifically:
1.  **Calculate Scores**: Calculate the accuracy (percentage) for three groups of questions:
    *   All questions (Total).
    *   Two specific questions related to "Refraction Angle" and "Double Rainbow" (Specific 2).
    *   The remaining 8 questions (Remaining 8).
2.  **Prepare Covariate**: Use the "proportion of time spent reading the original text" (calculated in previous steps) as a covariate.
3.  **Statistical Analysis**:
    *   Execute a mixed-design ANCOVA or a series of ANCOVAs where "Noticed Error" (T/F) is the between-subjects factor, the "Question Group" is the within-subjects factor (if applicable), and the "reading time proportion" is the covariate.
4.  **Reporting**: Output the results including F-statistics, p-values, confidence intervals, and effect sizes (e.g., partial eta squared). Provide a detailed interpretation of the findings in Japanese.

## calculate_scores_and_covariate

### Subtask:
Calculate accuracy scores for the three question groups and verify the covariate for the analysis.


In [13]:
import re

def extract_score(text):
    if pd.isna(text):
        return 0.0
    if isinstance(text, (int, float)):
        return float(text)
    # Matches patterns like '10.00 / 10' or just '10.00'
    match = re.search(r'(\d+\.?\d*)\s*/\s*\d+', str(text))
    if match:
        return float(match.group(1))
    match_simple = re.search(r'(\d+\.?\d*)', str(text))
    if match_simple:
        return float(match_simple.group(1))
    return 0.0

# Identify columns containing '[Score]'
score_cols = [col for col in df.columns if '[Score]' in col]

# Apply extraction to all score columns
for col in score_cols:
    df[f'{col}_numeric'] = df[col].apply(extract_score)

# Map specific questions based on their content keywords found in the column list
# 屈折角 related: '虹が「色の帯」として見えるのは...屈折角... [Score]'
# 二重虹 related: '「二重虹（外側の虹）」が形成される際... [Score]'
refraction_col = [c for c in score_cols if '屈折角' in c][0] + '_numeric'
double_rainbow_col = [c for c in score_cols if '二重虹' in c and '反射' in c][0] + '_numeric'

specific_2_cols = [refraction_col, double_rainbow_col]
remaining_8_cols = [f'{c}_numeric' for c in score_cols if f'{c}_numeric' not in specific_2_cols][:8]
final_10_cols = specific_2_cols + remaining_8_cols

# Calculate scores (assuming max score per question is 10, normalize to 0-1 range)
df['total_score'] = df[final_10_cols].mean(axis=1) / 10.0
df['specific_2_score'] = df[specific_2_cols].mean(axis=1) / 10.0
df['remaining_8_score'] = df[remaining_8_cols].mean(axis=1) / 10.0

print("Scores calculated successfully.")
print(f"Questions used: {len(final_10_cols)}")
display(df[['total_score', 'specific_2_score', 'remaining_8_score', 'original_reading_proportion']].head())

Scores calculated successfully.
Questions used: 10


,total_score,specific_2_score,remaining_8_score,original_reading_proportion
number,,,,
1,0.5,0.5,0.500,0.830547
2,0.7,0.5,0.750,0.653335
3,0.6,1.0,0.500,0.620690
4,0.4,0.5,0.375,0.620376
5,0.8,1.0,0.750,0.680591


## mixed_design_ancova_analysis

### Subtask:
Perform a mixed-design ANCOVA or separate ANCOVAs to analyze the effect of noticing errors on test scores, using the reading time proportion as a covariate.


In [14]:
try:
    import pingouin as pg
except ImportError:
    !pip install pingouin
    import pingouin as pg

# 1. ANCOVA for Specific 2 Score
print("--- ANCOVA for Specific 2 Score ---")
ancova_specific = pg.ancova(data=df, dv='specific_2_score', covar='original_reading_proportion', between='誤りを気づいた')
display(ancova_specific)

# 2. ANCOVA for Remaining 8 Score
print("\n--- ANCOVA for Remaining 8 Score ---")
ancova_remaining = pg.ancova(data=df, dv='remaining_8_score', covar='original_reading_proportion', between='誤りを気づいた')
display(ancova_remaining)

# 3. ANCOVA for Total Score
print("\n--- ANCOVA for Total Score ---")
ancova_total = pg.ancova(data=df, dv='total_score', covar='original_reading_proportion', between='誤りを気づいた')
display(ancova_total)

# Descriptive stats for context
print("\n--- Descriptive Statistics by Group ---")
desc_stats = df.groupby('誤りを気づいた')[['specific_2_score', 'remaining_8_score', 'total_score']].agg(['mean', 'std'])
display(desc_stats)

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 204.8/204.8 kB 5.0 MB/s eta 0:00:00
--- ANCOVA for Specific 2 Score ---


,Source,SS,DF,F,p_unc,np2
0,誤りを気づいた,0.065281,1,1.187635,0.304124,0.116576
1,original_reading_proportion,0.005292,1,0.096269,0.763418,0.010583
2,Residual,0.494708,9,NaN,NaN,NaN



--- ANCOVA for Remaining 8 Score ---


,Source,SS,DF,F,p_unc,np2
0,誤りを気づいた,0.061352,1,1.894528,0.201965,0.173897
1,original_reading_proportion,0.000213,1,0.006589,0.937081,0.000732
2,Residual,0.291453,9,NaN,NaN,NaN



--- ANCOVA for Total Score ---


,Source,SS,DF,F,p_unc,np2
0,誤りを気づいた,0.062128,1,2.266907,0.166425,0.201201
1,original_reading_proportion,0.000008,1,0.000299,0.986580,0.000033
2,Residual,0.246658,9,NaN,NaN,NaN



--- Descriptive Statistics by Group ---


specific_2_score       remaining_8_score           total_score  \
                    mean   std              mean       std        mean   
誤りを気づいた                                                                  
F               0.833333  0.25          0.625000  0.139754    0.666667   
T               1.000000  0.00          0.791667  0.260208    0.833333   

                   
              std  
誤りを気づいた            
F        0.141421  
T        0.208167

### ANCOVA Analysis Summary

I performed three separate ANCOVAs to evaluate the effect of noticing errors (Between-subjects factor: '誤りを気づいた') on test performance, while controlling for the proportion of time spent reading the original text (Covariate: 'original_reading_proportion').

#### 1. Specific 2 Questions (Refraction & Double Rainbow)
- **Effect of Noticing Error**: $F(1, 9) = 1.188$, $p = 0.304$, $\eta^2_p = 0.117$
- **Covariate (Time Proportion)**: $F(1, 9) = 0.096$, $p = 0.763$
- **Result**: Not statistically significant.

#### 2. Remaining 8 Questions
- **Effect of Noticing Error**: $F(1, 9) = 1.895$, $p = 0.202$, $\eta^2_p = 0.174$
- **Covariate (Time Proportion)**: $F(1, 9) = 0.007$, $p = 0.937$
- **Result**: Not statistically significant.

#### 3. Total Score (All 10 Questions)
- **Effect of Noticing Error**: $F(1, 9) = 2.267$, $p = 0.166$, $\eta^2_p = 0.201$
- **Covariate (Time Proportion)**: $F(1, 9) = 0.0003$, $p = 0.987$
- **Result**: Not statistically significant.

---

### 日本語による統計的解釈 (Statistical Interpretation in Japanese)

**分析結果の要約:**
共分散分析（ANCOVA）の結果、原文を読む時間の割合を制御した上でも、「誤りに気づいたかどうか」がテストの正答率（特定2問、残り8問、および合計点）に対して与える統計的に有意な影響は認められませんでした（いずれも $p > 0.05$）。

1.  **特定質問（屈折角・二重虹）**: 誤りに気づいた群の方が平均点が高い傾向にありますが（$M_{True}=1.0$ vs $M_{False}=0.83$）、サンプルサイズが小さいため統計的な有意差には至りませんでした（$p=0.304$）。
2.  **残り8問および合計点**: 同様に、誤りに気づいた群のスコアが高い傾向は見られるものの、統計的に有意な差ではありませんでした。
3.  **共変量の影響**: 原文を読む時間の割合も、テストスコアに対して有意な関連性を示しませんでした。

**結論:**
現時点のデータ（$N=12$）では、誤りに気づくことが直接的にテストパフォーマンスを向上させるとは結論付けられません。ただし、合計スコアにおける効果量（$\eta^2_p = 0.201$）は中程度から大程度の効果を示唆しており、サンプルサイズを拡大することで有意な差が検出される可能性があります。

## result_interpretation_jp

### Subtask:
Summarize the statistical findings from the logistic regression and ANCOVA analyses in Japanese, focusing on the hypothesis testing results.


## Summary:

### Q&A

**「誤りに気づいたかどうか」はテストの正答率に影響を与えましたか？**
統計的には、有意な影響は認められませんでした。共分散分析（ANCOVA）の結果、特定質問（$p = 0.304$）、残り8問（$p = 0.202$）、合計点（$p = 0.166$）のいずれにおいても $p > 0.05$ となりました。

**読解時間の割合（共変量）はテストスコアに関連していましたか？**
いいえ、原文を読む時間の割合は、テストスコアに対して統計的に有意な関連性を示しませんでした（すべてのモデルで $p > 0.70$）。

---

### Data Analysis Key Findings

*   **記述統計の傾向**: 統計的な有意差はないものの、平均スコアでは「誤りに気づいた群」が「気づかなかった群」を上回る傾向が見られました。
    *   合計スコア平均: 気づいた群 $0.83$ vs 気づかなかった群 $0.67$。
    *   特定2問（屈折角・二重虹）: 気づいた群 $1.0$ vs 気づかなかった群 $0.83$。
*   **効果量（Partial Eta Squared）**: 合計スコアにおける部分エータ二乗値（$\eta^2_p$）は $0.201$ でした。これは一般的に「大きな効果」の部類に入りますが、サンプルサイズが小さいため有意確率（$p$値）には反映されませんでした。
*   **ANCOVAの結果**:
    *   特定2問: $F(1, 9) = 1.188, p = 0.304$
    *   合計スコア: $F(1, 9) = 2.267, p = 0.166$

---

### Insights or Next Steps

*   **サンプルサイズの拡大**: 今回の分析は $N=12$ と非常に小規模です。合計スコアで中〜大程度の効果量（$\eta^2_p = 0.201$）が示唆されているため、サンプルサイズを増やすことで「誤りに気づくこと」による学習効果が有意に検出される可能性が高いと考えられます。
